### PR TITLE
feat: refactored yggd main action.

### PR DESCRIPTION
**Pull request by jirihnidek**
* Converted one huge function into several functions.
* The functionality should be almost the same.

### CODE REVIEW

**Review by subpop — CHANGES_REQUESTED** [review]
This looks great! I have a pattern question related to `systemdWatchDog`, but otherwise I like the clean up. Thank you!

File: cmd/yggd/main.go [hunk 1]
```diff
@@ -26,6 +27,373 @@ var (
...
+// systemdWatchDog tries to send sd_notify to systemd.
+// More details about sd_notify can be found here:
+// https://www.freedesktop.org/software/systemd/man/sd_notify.html
+func systemdWatchDog() {
+	watchdogDuration, err := daemon.SdWatchdogEnabled(false)
+	if err != nil {
+		log.Errorf("cannot get watchdog duration: %v", err)
+	}
+	if watchdogDuration > 0 {
+		go func() {
```
[the remark this answers is not in the document]
Comment (jirihnidek) [reply cmd/yggd/main.go]: I also like the pattern that of all these functions are running on goroutines, so I removed the `go func()` here. What do you think?

File: cmd/yggd/main.go [hunk 2]
```diff
@@ -26,6 +27,371 @@ var (
...
+// systemdWatchDog tries to send sd_notify to systemd.
+// More details about sd_notify can be found here:
+// https://www.freedesktop.org/software/systemd/man/sd_notify.html
+func systemdWatchDog() {
+	watchdogDuration, err := daemon.SdWatchdogEnabled(false)
+	if err != nil {
+		log.Errorf("cannot get watchdog duration: %v", err)
+	}
+	if watchdogDuration > 0 {
+		var sdState = daemon.SdNotifyWatchdog
```
Comment (subpop) [inline cmd/yggd/main.go:275]: Why is this assigned to a variable? It doesn't appear like this value will
change between the declaration and the final usage. Wouldn't it be cleaner to
just directly pass `daemon.SdNotifyWatchdog` to `daemon.SdNotify()`?

File: cmd/yggd/main.go [hunk 3]
```diff
@@ -26,6 +27,371 @@ var (
...
+
+// systemdWatchDog tries to send sd_notify to systemd.
+// More details about sd_notify can be found here:
+// https://www.freedesktop.org/software/systemd/man/sd_notify.html
+func systemdWatchDog() {
+	watchdogDuration, err := daemon.SdWatchdogEnabled(false)
+	if err != nil {
+		log.Errorf("cannot get watchdog duration: %v", err)
+	}
+	if watchdogDuration > 0 {
```
Comment (subpop) [inline cmd/yggd/main.go:274]: I feel like I want to see a debug message here.

```suggestion
	if watchdogDuration > 0 {
        log.Debug("starting systemd watchdog notification")
```

**Review by subpop — CHANGES_REQUESTED** [review]
This is great. I think I'd like to see a debug message when the watchdog notification loop starts, but otherwise, I'm ready to merge this.

File: cmd/yggd/main.go (hunk 2 above)
Comment (jirihnidek) [reply cmd/yggd/main.go]: Fixed.

File: cmd/yggd/main.go (hunk 3 above)
Comment (jirihnidek) [reply cmd/yggd/main.go:274]: Debug message added.

**Review by subpop — APPROVED** [review]

**Pull request merged**